### ISSUE TITLE
HTF-0.13.2.5 can't compile its test suite

### ISSUE THREAD

**Issue by peti** [skogsbaer/HTF#74]
I'm seeing the following error trying to build HTF with testing enabled:

~~~
Preprocessing test suite 'TestHTF' for HTF-0.13.2.5..
Building test suite 'TestHTF' for HTF-0.13.2.5..
ghc: could not execute: ./scripts/local-htfpp
~~~

**Comment by skogsbaer** [comment]
I cannot reproduce your problem (under Mac OSX). I do the following:

    tar xfz HTF-0.13.2.5.tar.gz
    cd HTF-0.13.2.5
    stack build --test

Can your provide some details? Operating system, build command and similar.

**Comment by peti** [comment]
~~~
$ runhaskell Setup.hs configure --enable-tests && runhaskell Setup.hs build && runhaskell Setup.hs test
Configuring HTF-0.13.2.5...
Preprocessing library for HTF-0.13.2.5..
Building library for HTF-0.13.2.5..
[ 1 of 23] Compiling Test.Framework.Colors ( Test/Framework/Colors.hs, dist/build/Test/Framework/Colors.o )
[ 2 of 23] Compiling Test.Framework.Diff ( Test/Framework/Diff.hs, dist/build/Test/Framework/Diff.o )
[ 3 of 23] Compiling Test.Framework.Location ( Test/Framework/Location.hs, dist/build/Test/Framework/Location.o )
[ 4 of 23] Compiling Test.Framework.Preprocessor ( Test/Framework/Preprocessor.hs, dist/build/Test/Framework/Preprocessor.o )
[ 5 of 23] Compiling Test.Framework.Pretty ( Test/Framework/Pretty.hs, dist/build/Test/Framework/Pretty.o )
[ 6 of 23] Compiling Test.Framework.Process ( Test/Framework/Process.hs, dist/build/Test/Framework/Process.o )
[ 7 of 23] Compiling Test.Framework.TestInterface ( Test/Framework/TestInterface.hs, dist/build/Test/Framework/TestInterface.o )
[ 8 of 23] Compiling Test.Framework.History ( Test/Framework/History.hs, dist/build/Test/Framework/History.o )
[ 9 of 23] Compiling Test.Framework.AssertM ( Test/Framework/AssertM.hs, dist/build/Test/Framework/AssertM.o )
[10 of 23] Compiling Test.Framework.TestTypes ( Test/Framework/TestTypes.hs, dist/build/Test/Framework/TestTypes.o )
[11 of 23] Compiling Test.Framework.JsonOutput ( Test/Framework/JsonOutput.hs, dist/build/Test/Framework/JsonOutput.o )
[12 of 23] Compiling Test.Framework.ThreadPool ( Test/Framework/ThreadPool.hs, dist/build/Test/Framework/ThreadPool.o )
[13 of 23] Compiling Test.Framework.Tutorial ( Test/Framework/Tutorial.hs, dist/build/Test/Framework/Tutorial.o )
[14 of 23] Compiling Test.Framework.Utils ( Test/Framework/Utils.hs, dist/build/Test/Framework/Utils.o )
[15 of 23] Compiling Test.Framework.QuickCheckWrapper ( Test/Framework/QuickCheckWrapper.hs, dist/build/Test/Framework/QuickCheckWrapper.o )
[16 of 23] Compiling Test.Framework.PrettyHaskell ( Test/Framework/PrettyHaskell.hs, dist/build/Test/Framework/PrettyHaskell.o )
[17 of 23] Compiling Test.Framework.HUnitWrapper ( Test/Framework/HUnitWrapper.hs, dist/build/Test/Framework/HUnitWrapper.o )
[18 of 23] Compiling Test.Framework.XmlOutput ( Test/Framework/XmlOutput.hs, dist/build/Test/Framework/XmlOutput.o )
[19 of 23] Compiling Test.Framework.TestReporter ( Test/Framework/TestReporter.hs, dist/build/Test/Framework/TestReporter.o )
[20 of 23] Compiling Test.Framework.CmdlineOptions ( Test/Framework/CmdlineOptions.hs, dist/build/Test/Framework/CmdlineOptions.o )
[21 of 23] Compiling Test.Framework.TestManager ( Test/Framework/TestManager.hs, dist/build/Test/Framework/TestManager.o )
[22 of 23] Compiling Test.Framework.BlackBoxTest ( Test/Framework/BlackBoxTest.hs, dist/build/Test/Framework/BlackBoxTest.o )
[23 of 23] Compiling Test.Framework   ( Test/Framework.hs, dist/build/Test/Framework.o )
Preprocessing test suite 'TestThreadPools' for HTF-0.13.2.5..
Building test suite 'TestThreadPools' for HTF-0.13.2.5..
[1 of 1] Compiling Main             ( tests/ThreadPoolTest.hs, dist/build/TestThreadPools/TestThreadPools-tmp/Main.o )
Linking dist/build/TestThreadPools/TestThreadPools ...
Preprocessing test suite 'MiscTests' for HTF-0.13.2.5..
Building test suite 'MiscTests' for HTF-0.13.2.5..
[1 of 1] Compiling Main             ( tests/MiscTest.hs, dist/build/MiscTests/MiscTests-tmp/Main.o )
Linking dist/build/MiscTests/MiscTests ...
Preprocessing test suite 'TestHTF' for HTF-0.13.2.5..
Building test suite 'TestHTF' for HTF-0.13.2.5..
./scripts/local-htfpp: runInteractiveProcess: runInteractiveProcess: exec: permission denied (Permission denied)
$ ls -l scripts/local-htfpp
-rw-r--r-- 1 simons users 927 Apr  1  2018 scripts/local-htfpp
~~~

That's building with ghc-8.4.4 on NixOS Linux/x86_64.

**Comment by skogsbaer** [comment]
Fixed in master with #76
